sig/steep/project/dsl.rbs

```diff
@@ -24,9 +24,13 @@ module Steep
 
         attr_reader project: Project?
 
-        attr_reader collection_config_path: Pathname?
-
-        NONE: untyped
+        # Attribute to keep track of collection configuration
+        #
+        # * `Pathname` means loading the configuration from the path
+        # * `nil` means no configuration is given
+        # * `false` means rbs-collection is disabled
+        #
+        attr_reader collection_config_path: Pathname | nil | false
```

Comment:
```suggestion
        # Attribute to keep track of collection configuration
        #
        # * `Pathname` means loading the configuration from the path
        # * `nil` means no configuration is given
        # * `false` means rbs-collection is disabled
        #
        attr_reader collection_config_path: Pathname? | false
```

Helps convey that `Pathname` and `nil` are similar (both about explicitly given configs) while `false` is unique.